kdump/include/runtest.sh

```diff
@@ -511,7 +506,82 @@ AppendConfig()
         shift
     done
 
-    rstrnt-report-log -l "${KDUMP_CONFIG}"
+    RhtsSubmit "${KDUMP_CONFIG}"
+}
+
+
+AppendSysconfig()
+{
+    Log "- Modifying /etc/sysconfig/kdump"
+
+    local KEY=$1
+    local ACTION=$2
+    local VALUE1=$3
+    local VALUE2=$4
+
+    if [ -z "$KEY" ] || [ -z "$ACTION" ] || [ -z "$VALUE1" ]; then
+        Error "- Missing KEY or ACTION or VALUE1."
+        return 1
+    elif [ "$ACTION" = "replace" ] && [ -z "$VALUE2" ]; then
+        Error "- Missing new_value for replacing."
+        return 1
+    elif ! grep "^$KEY=\"" "${KDUMP_SYS_CONFIG}"; then
+        Error "- Invalid KEY: $KEY."
+        return 1
+    fi
+
+    local kdump_sys_config_tmp="${KDUMP_SYS_CONFIG}.tmp"
+    \cp "${KDUMP_SYS_CONFIG}" "${kdump_sys_config_tmp}"
+
+    # Note: When KEY is "KDUMP_COMMANDLINE", ACITION add/remove/replace is actually
+    # manipulating the value of current kernel cmdline and assign
+    # it to KDUMP_COMMANDLINE.
+    # if there is no value set to KDUMP_COMMANLINE, assign current kernel cmdline
+    # to it for later string manipluation.
+    if [ "$KEY" == "KDUMP_COMMANDLINE" ] && \
+        grep "^$KEY=\"[[:space:]]*\"" "${kdump_sys_config_tmp}"; then
+
+        sed -i "/^KDUMP_COMMANDLINE=\"/d" "${kdump_sys_config_tmp}"
+        echo "KDUMP_COMMANDLINE=\"$(cat /proc/cmdline)\"" >> "${kdump_sys_config_tmp}"
```

Comment:
Why not replace with `sed` in one step, instead of removing the line first and then add it at the end?

Reply:
sed replace in one stop may look good. But it's more straight forwad to simply delete the old line and add the correct line back.